lambda.tf

```
@@ -0,0 +1,68 @@
+# ------------------------------------------------------------------------------
+# The AWS Lambda function that is used to publish egress IP addresses.
+# The ZIP file is created with:
+# http://github.com/cisagov/publish-egress-ip-lambda
+# ------------------------------------------------------------------------------
+
+resource "aws_lambda_function" "publish_egress_ip" {
+  provider = aws.deploy
+
+  filename         = var.lambda_zip_filename
+  source_code_hash = filebase64sha256(var.lambda_zip_filename)
+  function_name    = var.lambda_function_name
+  role             = aws_iam_role.lambdaexecution_role.arn
+  handler          = "lambda_handler.handler"
+  runtime          = "python3.9"
+  timeout          = 600
+  memory_size      = 128
+  description      = var.lambda_function_description
+}
+
+# The CloudWatch log group for the Lambda function
+resource "aws_cloudwatch_log_group" "lambda_logs" {
+  provider = aws.deploy
+
+  name              = format("/aws/lambda/%s", var.lambda_function_name)
+  retention_in_days = 30
+}
+
+# Schedule the Lambda function to run every X minutes
+resource "aws_cloudwatch_event_rule" "lambda_schedule" {
+  provider = aws.deploy
+
+  description         = format("Executes %s Lambda every %s minutes", var.lambda_function_name, var.lambda_schedule_interval)
+  name                = format("%s-every-%s-minutes", var.lambda_function_name, var.lambda_schedule_interval)
+  schedule_expression = format("rate(%s minutes)", var.lambda_schedule_interval)
```

Use `%d` instead of `%s` to correspond to `var.lambda_schedule_interval` becoming a number:
```suggestion
  description         = format("Executes %s Lambda every %d minutes", var.lambda_function_name, var.lambda_schedule_interval)
  name                = format("%s-every-%d-minutes", var.lambda_function_name, var.lambda_schedule_interval)
  schedule_expression = format("rate(%d minutes)", var.lambda_schedule_interval)
```

---

So a note here is that this controls what the value is converted to per the [documentation](https://www.terraform.io/language/functions/format#verbs) and does not specify the provided type. We should use `%d` anywhere we expect/need an integer value (or whatever the appropriate `%` verb is for what we expect/need) no matter what type the source is to ensure behavior. Examples from the console:

```console
> format("%s %d", "60.15", "60")
"60.15 60"
> format("%s %d", 60.15, 60.15)
╷
│ Error: Error in function call
│
│   on <console-input> line 1:
│   (source code not available)
│
│ Call to function "format" failed: unsupported value for "%d" at 3: an integer is required.
╵


> format("%s %d", 60, 60)
"60 60"
> format("%s %d", 60, "60")
"60 60"
```

---

It's a little bit ugly (if you have a better solution, please let me know), but take a look at the validation I added in https://github.com/cisagov/publish-egress-ip-terraform/pull/6/commits/b4755aeeafc683a5cf6abb0294aa92b7d1609446.